src/applications/virtual-agent/tests/components/webchat/startConvoAndTrackUtterances.unit.spec.js

```diff
@@ -0,0 +1,199 @@
+
+    expect(actions[0].payload.activity).to.own.include({
+      name: 'startConversation',
+    });
+    expect(actions[0]).to.own.include({
+      type: 'DIRECT_LINE/POST_ACTIVITY',
+    });
+  });
+
+  it('should correctly handle "DIRECT_LINE/CONNECT_FULFILLED" with auth false', async () => {
+    await StartConvoAndTrackUtterances.makeBotStartConvoAndTrackUtterances(
+      'csrfToken',
+      'apiSession',
+      'apiURL',
+      'baseURL',
+      'userFirstName',
+      'userUuid',
+      false,
+    )(store)(fakeNext)(connectFulfilledAction);
+
+    const actions = store.getActions();
+    expect(actions.length).to.equal(1);
+    expect(fakeNext.callCount).to.equal(1);
+
+    expect(actions[0]).to.own.include({ type: 'WEB_CHAT/SEND_EVENT' });
+    expect(actions[0].payload).to.own.include({ name: 'webchat/join' });
+  });
+
+  describe('Handling of "DIRECT_LINE/INCOMING_ACTIVITY"', () => {
+    const IS_TRACKING_UTTERANCES = 'va-bot.isTrackingUtterances';
+    const IN_AUTH_EXP = 'va-bot.inAuthExperience';
+    const RECENT_UTTERANCES = 'va-bot.recentUtterances';
+    const sandbox = sinon.createSandbox();
+    beforeEach(() => {
+      sessionStorage.clear();
+    });
+    afterEach(() => {
+      sessionStorage.clear();
+      sandbox.restore();
+    });
+    it("should correctly begin tracking utterances if it hasn't yet", async () => {
+      // setup
+      // fire/execute
+      await StartConvoAndTrackUtterances.makeBotStartConvoAndTrackUtterances(
+        'csrfToken',
+        'apiSession',
+        'apiURL',
+        'baseURL',
+        'userFirstName',
+        'userUuid',
+        true,
+      )(store)(fakeNext)(directIncomingActivity);
+      // tests
+      const isTrackingUtterances = await sessionStorage.getItem(
+        IS_TRACKING_UTTERANCES,
+      );
+      expect(isTrackingUtterances).to.equal('true');
+    });
+    it('Stops tracking utterances when about to redirect to sign in', async () => {
+      // setup
+      const activity = {
+        type: 'message',
+        text: 'Alright. Sending you to the sign in page...',
+        from: { role: 'bot' },
+      };
+      const aboutToSignInActivity = {
+        type: 'DIRECT_LINE/INCOMING_ACTIVITY',
+        payload: { activity },
+      };
+      const spyDispatchEvent = sandbox.spy(window, 'dispatchEvent');
+
+      // fire
+      await StartConvoAndTrackUtterances.makeBotStartConvoAndTrackUtterances(
+        'csrfToken',
+        'apiSession',
+        'apiURL',
+        'baseURL',
+        'userFirstName',
+        'userUuid',
+        true,
+      )(store)(fakeNext)(aboutToSignInActivity);
+      // tests
+      const isTrackingUtterances = await sessionStorage.getItem(
+        IS_TRACKING_UTTERANCES,
+      );
+      expect(sessionStorage.length).to.equal(1);
+      expect(isTrackingUtterances).to.equal('false');
+      expect(spyDispatchEvent.callCount).to.equal(1);
+
+      expect(spyDispatchEvent.firstCall.args[0].data).to.equal(activity);
```

Comment:
Suggestion - add expectation that event should be a `webchat-auth-activity` event. Another component (Chatbox) expects to receive this event, so to be thorough, we can validate this contract is met in the test